src/careamics/config/training_model.py

```diff
@@ -37,6 +37,20 @@ class TrainingConfig(BaseModel):
     num_epochs: int = Field(default=20, ge=1)
     """Number of epochs, greater than 0."""
 
+    precision: Literal["64", "32", "16-mixed", "bf16-mixed"] = Field(default="32")
+    """Numerical precision"""
+    max_steps: int = Field(default=-1, ge=-1, exclude=0)
```

Comment:
`exclude` excludes the field from model dumps (see [pydantic notes](https://docs.pydantic.dev/latest/concepts/fields/#exclude)), it does not exclude values.

You will need a field validator here.

```python
from pydantic import BaseModel, Field

class Stuff(BaseModel):
    max_steps: int = Field(default=-1, ge=-1, exclude=0)

Stuff(max_steps=0)
```